File: packages/http-client-java/generator/http-client-generator-test/src/test/java/documentation/DocumentationTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package documentation;

import documentation.lists.models.BulletPointsEnum;
import documentation.lists.models.BulletPointsModel;
import org.junit.jupiter.api.Test;

// Markdown in Javadoc is not supported till JDK23
public class DocumentationTests {

    private final ListsClient listsClient = new DocumentationClientBuilder().buildListsClient();
    private final TextFormattingClient textFormattingClient
        = new DocumentationClientBuilder().buildTextFormattingClient();

    @Test
    public void testBulletPointsOp() {
        listsClient.bulletPointsOp();
    }

    @Test
    public void testBulletPointsModel() {
        listsClient.bulletPointsModel(new BulletPointsModel(BulletPointsEnum.SIMPLE));
    }

    @Test
    public void testNumbered() {
        listsClient.numbered();
    }

    @Test
    public void testBoldText() {
        textFormattingClient.boldText();
    }

    @Test
    public void testItalicText() {
        textFormattingClient.italicText();
    }

    @Test
    public void testCombinedFormatting() {
        textFormattingClient.combinedFormatting();
    }
}

File: packages/http-client-csharp/generator/TestProjects/Spector.Tests/Http/Documentation/DocumentationTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Linq;
using System.Reflection;
using System.Threading;
using System.Threading.Tasks;
using System.Xml;
using Documentation;
using Documentation._Lists;
using Documentation._TextFormatting;
using NUnit.Framework;

namespace TestProjects.Spector.Tests.Http.Documentation
{
    public class DocumentationTests : SpectorTestBase
    {
        // Lists namespace tests
        [SpectorTest]
        public Task BulletPointsOp() => Test(async (host) =>
        {
            var client = new DocumentationClient(host, new DocumentationClientOptions());
            var response = await client.GetListsClient().BulletPointsOpAsync();
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });

        [SpectorTest]
        [Ignore("https://github.com/microsoft/typespec/issues/9173")]
        public Task BulletPointsModel() => Test(async (host) =>
        {
            var client = new DocumentationClient(host, new DocumentationClientOptions());
            var input = new BulletPointsModel(BulletPointsEnum.Simple);
            var response = await client.GetListsClient().BulletPointsModelAsync(input);
            Assert.AreEqual(200, response.GetRawResponse().Status);
        });

        [SpectorTest]
        public Task Numbered() => Test(async (host) =>
        {
            var client = new DocumentationClient(host, new DocumentationClientOptions());
            var response = await client.GetListsClient().NumberedAsync();
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });

        // TextFormatting namespace tests
        [SpectorTest]
        public Task BoldText() => Test(async (host) =>
        {
            var client = new DocumentationClient(host, new DocumentationClientOptions());
            var response = await client.GetTextFormattingClient().BoldTextAsync();
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });

        [SpectorTest]
        public Task ItalicText() => Test(async (host) =>
        {
            var client = new DocumentationClient(host, new DocumentationClientOptions());
            var response = await client.GetTextFormattingClient().ItalicTextAsync();
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });

        [SpectorTest]
        public Task CombinedFormatting() => Test(async (host) =>
        {
            var client = new DocumentationClient(host, new DocumentationClientOptions());
            var response = await client.GetTextFormattingClient().CombinedFormattingAsync();
            Assert.AreEqual(204, response.GetRawResponse().Status);
        });
    }
}
